### DISCUSSION TITLE
Problema con ejecucion de .jar

### DISCUSSION THREAD

**Discussion by brianstremel** [dds-jv/foro#143] [Consultas de herramientas] [answered]
Buenas, estoy intentando ejecutar los .jar pero no me estaría dejando.
Primero por consola ejecuto "mvn clean package" y me crea 2 archivos:
1-TPAnual-1.0-SNAPSHOT.jar
2-TPAnual-1.0-SNAPSHOT-jar-with-dependencies.jar
Los intento ejecutar con java -jar y me tira los siguiente
![image](https://user-images.githubusercontent.com/37552405/84799007-0465e100-afd2-11ea-93a7-88531dce2203.png)

El Main lo tengo puesto de la siguiente forma:
![image](https://user-images.githubusercontent.com/37552405/84799309-7b02de80-afd2-11ea-9db1-f0d34e0176b8.png)
![image](https://user-images.githubusercontent.com/37552405/84799353-8c4beb00-afd2-11ea-9d7f-59b3739b5313.png)

Pensé que podría haber sido el package y lo cambié por package ar.edu.frba.gesoc;
pero directamente no encontró nada:
![image](https://user-images.githubusercontent.com/37552405/84799606-e0ef6600-afd2-11ea-8ec7-aab74e60d45d.png)

Y este es el pom:
![image](https://user-images.githubusercontent.com/37552405/84799901-54917300-afd3-11ea-92dd-50de00ae1004.png)

En donde le estaría pifiando? 

Edit: agrego los errores por escrito:
no main manifest attribute,  in TPAnual-1.0-SNAPSHOT.jar
Error:  Could not find or load main class ar.edu.frba.dds.gesoc.Main

**Comment by gastonprieto** [comment]
Hola!

Fijate que tenes dos errores que son distintos en las capturas que mandaste,  uno es que hay algo con el manifest y el otro es que no esta el archivo jar creo. 

En uno ejecutaría un `ls` para ver si se creo el archivo o si tiene otro nombre.
Y en la otra me aseguraría que el path este bien y que el main cumpla la firma que te pide Java para que sea un punto de entrada.

Por otro lado trata de no subir fotos (o al menos no fotos del error), porque si alguien después tiene el mismo error no lo va a poder encontrar en el buscador.

**Comment by FeliIsra** [comment]
Hola ! Nosotros tenemos el mismo error que puso Brian al principio y no podemos arreglarlo.

Error: no se ha encontrado o cargado la clase principal src.main.java.Main.Main

Alguna idea ?

Gracias !

**Comment by DiegoGuida** [comment]
Hola! nosotros teníamos el mismo error y lo solucionamos corrigiendo el manifest en el pom. El cual tiene que coincidir con la ubicacion del archivo Main.
En el caso de la foto, la ruta del main es: ar.edu.frba.gesoc.Main y en el manifest del pom esta escrito: ar.edu.frba.dds.gesoc.Main 
Por lo cual, nunca te encuentra el main class.

**Comment by brianstremel** [comment, marked as answer]
@DiegoGuida genial, ese era el problema, tengo que admitir que incluso tuve que leer lo que escribiste varias veces para encontrar la diferencia. Gracias!
